### ISSUE TITLE
Question about potentially odd imageresolver package `getName` behaviour

### ISSUE THREAD

**Issue by greglanthier** [operator-framework/operator-manifest-tools#27]
First let me thank y'all for releasing **operator-manifest-tools**.

It is a very handy and useful addition to our build pipeline toolbox - and it really helped me get a handle on some odd behaviour I was seeing when I was experimenting with `operator-sdk generate bundle -q --overwrite --use-image-digests ...` while getting ready for air-gap enabling an operator I've been working on.

**The problem**

I think the **imageresolver** package is not dealing with image references that contain registry server hostnames that include non-default port numbers correctly.

**What I'm seeing now**

```command
$ ./bin/operator-manifest-tools pinning extract bundle/manifests | ./bin/operator-manifest-tools pinning resolve -
{"localhost:5005/controller:v0.0.1":"localhost@sha256:fca9baeb6abdd5d21c8350ca143aa1a778fbfed7adefeb79db95a181752cc323"}
$
```

**What I'm expecting to see**

```command
$ ./bin/operator-manifest-tools pinning extract bundle/manifests | ./bin/operator-manifest-tools pinning resolve -
{"localhost:5005/controller:v0.0.1":"localhost:5005/controller@sha256:fca9baeb6abdd5d21c8350ca143aa1a778fbfed7adefeb79db95a181752cc323"}
$
```

**Test case**

I used [operator-sdk v1.22.2](https://github.com/operator-framework/operator-sdk/releases/tag/v1.22.2) to generate an Ansible-based operator and used [ctlptl](https://github.com/tilt-dev/ctlptl/) to startup a kind cluster that is connected to a local registry server hosted at localhost:5005.

I ran the following command to build my operator, bundle, and catalog images & confirmed that they were pushed to the local registry properly.

```command
$ IMG=localhost:5005/controller:v0.0.1 make -f Makefile docker-build docker-push bundle bundle-build bundle-push catalog-build catalog-push
$ skopeo inspect --no-tags docker://localhost:5005/controller:v0.0.1
{
    "Name": "localhost:5005/controller",
    "Digest": "sha256:fca9baeb6abdd5d21c8350ca143aa1a778fbfed7adefeb79db95a181752cc323",
    "RepoTags": [],
    "Created": "2022-08-05T00:01:57.301078865Z",
    "DockerVersion": "",
    "Labels": { ...
}
```

I then used **operator-manifest-tools** 0.2.1 to attempt try to resolve the image digests for the images contained in the **bundle/manifests** directory of the operator source like this:

```command
$ ./bin/operator-manifest-tools pinning extract bundle/manifests | ./bin/operator-manifest-tools pinning resolve -
{"localhost:5005/controller:v0.0.1":"localhost@sha256:fca9baeb6abdd5d21c8350ca143aa1a778fbfed7adefeb79db95a181752cc323"}
$
```

I _think_ the problem is [here](https://github.com/operator-framework/operator-manifest-tools/blob/ee92fbc83da525980ee58c1ff28fd10d6a5f8416/pkg/imageresolver/imageresolver.go#L89-L95):
https://github.com/operator-framework/operator-manifest-tools/blob/ee92fbc83da525980ee58c1ff28fd10d6a5f8416/pkg/imageresolver/imageresolver.go#L89-L95

The grammar for Docker image references ([link](https://pkg.go.dev/github.com/docker/distribution/reference)) mentions the following:

```ebnf
reference                       := name [ ":" tag ] [ "@" digest ]
name                            := [domain '/'] path-component ['/' path-component]*
domain                          := domain-component ['.' domain-component]* [':' port-number]
domain-component                := /([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])/
port-number                     := /[0-9]+/
```

Is it reasonable to expect **operator-manifest-tools** to render a result that looks like this:
`{"localhost:5005/controller:v0.0.1":"localhost:5005/controller@sha256:fca9baeb6abdd5d21c8350ca143aa1a778fbfed7adefeb79db95a181752cc323"}` ?

**Comment by zach-source** [comment]
Hrm, this is likely a bug and shouldn't be. Let me see how we can patch it.

**Comment by zach-source** [comment]
Should be fixed with v0.2.2 @greglanthier
